liquigraph-core/src/main/java/org/liquigraph/core/api/MigrationRunner.java

```diff
@@ -90,6 +98,22 @@ public void runMigrations(Configuration configuration) {
         return persistedChangesets;
     }
 
+    private void replaceChecksum(Configuration configuration,
```

Comment:
I think this checksum replacement would be better in the `ChangelogGraphReader` class
It may look counter-intuitive to perform writes in a `XxxReader` class, but it already does to perform internal migrations.
In a way, `ChangelogGraphReader` output invariant is that it always returns a checksummed sequence of persisted migrations.

Reply:
> I think this checksum replacement would be better in the ChangelogGraphReader

I agree 100%.

WDYT about creating a ChangelogGraphReader interface that will be implemented both by the current ChangelogGraphReader and ReplaceChecksumGraphReader ?

ReplaceChecksumGraphReader will the delegate to ChangelogGraphReader to retrieve the changesets from db, and will check if there are some with null checksum, if that's the case it will update the corresponding changesets in db.

So to resume, we'll have something like

```

    private ChangelogReader changelogReader(Collection<Changeset> declaredChangesets){
      return new ReplaceChecksumGraphReader(declaredChangesets, changelogReader);
    }

    private Collection<Changeset> readPersistedChangesets(Collection<Changeset> declaredChangesets, Connection writeConnection) {
        ChangelogGraphReader reader = changelogReader(declaredChangesets);
        Collection<Changeset> persistedChangesets = reader.read(writeConnection);
       ... as before...
    }
```

Reply:
done in commit 9754dceb09ff62ed1b67b7522c0dd7b681e48feb

Reply:
I understand better now what you intended to do, thanks for taking the time to lay out your ideas.

I'm not 100% convinced decorator is a good fit here, as I do not need to compose different changelog reading strategies. Indeed, changesets without checksums should _always_ have their checksums re-computed, it may be easier to implement that part directly in the existing `ChangelogGraphReader`.

Reply:
done in commit 167e6de